src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -106,8 +106,8 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
                     if (dbPerm == null || Boolean.TRUE.equals(dbPerm.getDummy())) {
                       return Future.succeededFuture();
                     }
-                    if (dbPerm != null && Boolean.FALSE.equals(dbPerm.getMutable())
-                        && PermissionUtils.equals(perm, dbPerm)) {
+                    if (Boolean.TRUE.equals(dbPerm.getDeprecated()) || (Boolean.FALSE.equals(dbPerm.getMutable())
+                        && PermissionUtils.equals(perm, dbPerm))) {
```

Comment:
Note `PermissionUtils.equals(perm, dbPerm)` will compare perm display name as well. What's the chance that the new perm will also have `(Deprecated)` as prefix?

Reply:
@adamdickmeiss @craigmcnally Guys, shouldn't all comparison be done with the permission name (not displayName)?

Reply:
@hjiebsco .. Doesn't matter.. It will fail regardless of display name and my test shows it.  This condition should *not* use PermissionUtils.equals as this is not relevant. Either there's a colliision because a permission is defined elsewhere or NOT.

It will be utterly confusing if we allowed same name in two modules and .. suddenly because you change the definition in one of the modules there *might* be a conflict. I say *might* because it might be depending on the order of insertion because the context only allows ONE module to be attached to a permission.

Reply:
The permissionName is already the same.. The check can be simplified to:
```java
  if (Boolean.TRUE.equals(dbPerm.getDeprecated()) || Boolean.FALSE.equals(dbPerm.getMutable())) {
```

Reply:
I will mark this as resolved. But we still need to decide the implication of re-use to downgrade.